protostar/starknet/data_transformer.py

```diff
@@ -92,3 +95,24 @@ def transform(data: CairoData) -> PythonData:
             raise DataTransformerException(str(ex)) from ex
 
     return transform
+
+
+async def transform_calldata_to_cairo_data(
+    contract_class: ContractClass,
+    function_name: str,
+    calldata: Optional[CairoOrPythonData] = None,
+) -> CairoData:
+    if isinstance(calldata, collections.Mapping):
+        assert contract_class.abi, "Abi required but not found"
```

Comment:
I feel like this should also be `DataTransformerException` but i'm not completely sure of all the cases on this one